syntax = "proto2";

package Blocksettle.Storage;


message WalletBackupFile
{
   optional string description = 1;
   required string id = 2;

   required bytes seed1 = 3;
   required bytes seed2 = 4;

   required bytes chainCode1 = 5;
   required bytes chainCode2 = 6;

   optional string name = 7;
}

// used for signed transactions
message TXFile
{
   repeated bytes transactions         = 1;
}

message UTXOFile
{
   repeated Transactions txByWallet = 1;

   message Transactions {
      required string wallet = 1;
      repeated bytes  transactions = 2;
   }
}


enum ResponseEntryType
{
   CCTokenEntryType = 1;
   AuthAddressVerificationTransactionType = 3;
   AuthRevokeTransactionType = 4;
   CCDistributionTransactionType = 5;
   AuthAddressVerificationRejectType = 6;
   FundCCGenAddrTransactionType = 7;
}

message ResponseEntry
{
   required ResponseEntryType type = 1;
   required string            request_id = 2;
   required bytes             data = 3;
}

message ResponseFile
{
   repeated ResponseEntry entries = 1;
}


message CCTokenEntry {
   required string   userId = 1;
   required string   ccToken = 2;
   required string   ccProduct = 3;
   required uint32   amount = 4;
   required string   request_id = 5;
}

message AuthAddressSubmitMixIn {
   required float    fee = 1;
   required uint64   txValue = 2;
   required string   auth_address = 3;
}

message AuthAddressReservationInfo
{
   required string address = 1;
   required string user_name = 2;
   required string request_id = 3;
}

message AuthAddressVerificationTransaction {
   repeated AuthAddressReservationInfo reservations = 1;
   required bytes verification_transaction = 2;
}

message AuthRevokeTransaction
{
   repeated AuthAddressReservationInfo reservations = 1;
   required bytes revoke_transaction = 2;
}

message CCDistributionTransaction
{
   required string   cc_product = 1;
   repeated string   address = 2;
   required bytes    transaction = 3;
   repeated string   user_name = 4;
   repeated uint32   amount = 5;
   repeated string   request_id = 6;
}

message AuthVerificationRejectedMessage
{
   required string user_name = 1;
   required string auth_address = 2;
   required string request_id = 3;
}

message FundCCGenAddrTransaction
{
   required bytes    transaction = 1;
   repeated string   cc_product = 2;
   repeated uint32   amount = 3;
   repeated string   reference = 4;
}

message CCDefinitions
{
   required bytes    response = 1;
   required bytes    signature = 2;
}
